### PR TITLE
[BugFix] Support show full fields syntax.

### LINKED ISSUE #17233: show full fields from 'table' it not supported
https://dev.mysql.com/doc/refman/8.0/en/show-columns.html
### Steps to reproduce the behavior (Required)
1. show full fields from 'table'

### Expected behavior (Required)
same as show full columns from 'table'
### Real behavior (Required)
syntax error
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
2.4,2.5

### PR DESCRIPTION
**Pull request by Astralidea**
Signed-off-by: Astralidea <astralidea@163.com>

## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #17233

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 2.5
  - [x] 2.4
  - [x] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Review by gsashish-dev — DISMISSED** [review]

**Comment by Astralidea** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4 [hunk 1]
```diff
@@ -125,6 +125,7 @@ EXTERNAL: 'EXTERNAL';
 EXTRACT: 'EXTRACT';
 EVERY: 'EVERY';
 FALSE: 'FALSE';
+FIELDS: 'FIELDS';
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/StarRocksLex.g4:128]: add fields to noreserved

**Review by ABingHuang — DISMISSED** [review]
LGTM

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by HangyuanLiu — APPROVED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Astralidea** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.4

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.3

**Pull request merged**